### DISCUSSION TITLE
Is There a Different Technique for the "Read Wildcards in Order" Function?

### DISCUSSION THREAD

**Discussion by DavidDragonsage** [metercai/SimpleSDXL#69] [Q&A]
I regularly test my wildcards and base models using the combination, "Disable Seed Increment" and "Read Wildcards in Order". While testing Flux in SimpleSDXL today I realized that the "Read Wildcards in Order" option is missing.

However, I know that SimpleSDXL has advanced wildcard functions - which I do not understand yet. I suspect that these functions can probably achieve what I want, and this is probably why "Read Wildcards in Order" is not available.

How would I achieve the "Read Wildcards in Order" function in SimpleSDXL?

This prompt worked to display the __melanin__ wildcard sequentially, a wildcard containing 14 entries:

"a [__melanin__:L14:1] woman walks through the city with a gentle smile, sharp background, deep depth of field, smooth skin" I did use the two underscores on either side of the wildcard but this forum does not display them - instead it displays the wildcard in a bold font.

This technique did work properly, except it stripped the parentheses from the wildcard entries. For example, the first entry was supposed to be "(very dark skinned:1.4)" but instead the prompt became "very dark skinned:1.4". So "skinned" was the only word that was weighted instead of the whole wildcard entry "very dark skinned".

If I just used the melanin wildcard without the sequential function it worked properly. For example: "a __melanin__ woman walks through the city with a gentle smile, sharp background, deep depth of field, smooth skin" weighted the whole wildcard properly, as in "(very dark skinned:1.4)" for example.

**Comment by metercai** [comment]
`a [__melanin__:L14:1] woman walks through the city with a gentle smile, sharp background, deep depth of field, smooth skin`

`[__melanin__:L14:1]` The meaning of this is to take 14 consecutive lines in order from the first one in the wildcard file: melanin.txt

You can send me your melatonin sample for testing the following questions. Please indicate the current sample and actual output effect, as well as your expected output effect.

**Comment by DavidDragonsage** [comment]
Yes, of course. I was not being very clear. I have modified that wildcard for use with Flux Schnell, attached. The wildcard entries are shown in bold.

SimpleSDXL Prompt: "a [__melanin-flux__:L9:1] woman walks through the city with a gentle smile, sharp background, deep depth of field, smooth skin"
Negative: "blurry background, bokeh, __nsfw__"
Styles: Fooocus V2, Fooocus Enhance
Aspect Ratio: 896×1152 ∣ 7:9
Image Number, Flux: 9
Sharpness: 2
Seed: 9057272855973270044
Disable Seed Increment: checked

Processed prompt for the first image (Image #1 attached) was: "a very dark ebony melanin:1.4 woman walks through the city with a gentle smile, sharp background, deep depth of field, smooth skin"

I created a very short wildcard called melanin-test with just the first entry from melanin-flux (test wildcard attached)

Then ran this prompt: 
"a __melanin-test__ woman walks through the city with a gentle smile, sharp background, deep depth of field, smooth skin"

The processed prompt was what it should be (Image #2 attached): "a (very dark ebony melanin:1.4) woman walks through the city with a gentle smile, sharp background, deep depth of field, smooth skin"

In the first case the wildcard processor gave the 1.4 weighting to only the last word of the wildcard entry.
In the second case the whole phrase was weighted properly.

[melanin-flux.txt](https://github.com/user-attachments/files/16986173/melanin-flux.txt)


[melanin-test.txt](https://github.com/user-attachments/files/16986962/melanin-test.txt)


![Image#1 2024-09-12_16-22-02_8789](https://github.com/user-attachments/assets/77a7e7b1-430d-4c8c-b26d-866985e16957)

![Image#2 2024-09-12_16-53-50_8789](https://github.com/user-attachments/assets/b623aebc-456b-4eff-8f9c-668f293bd316)
